internal/checker/utilities.go

```diff
@@ -2044,7 +2043,7 @@ var getFeatureMap = sync.OnceValue(func() map[string][]FeatureMapEntry {
 })
 
 func rangeOfTypeParameters(sourceFile *ast.SourceFile, typeParameters *ast.NodeList) core.TextRange {
-	return core.NewTextRange(typeParameters.Pos()-1, min(len(sourceFile.Text), scanner.SkipTrivia(sourceFile.Text, typeParameters.End())+1))
+	return core.NewTextRange(max(0, typeParameters.Pos()-1), min(len(sourceFile.Text), scanner.SkipTrivia(sourceFile.Text, typeParameters.End())+1))
```

Comment:
How likely are we to hit this bug elsehwere when working with positions on synthetic nodes? IIRC we would never have to do this previously, but now there's a reason to check for each error.

Reply:
The actual problem is that typeParameters could never be parsed at the very beginning of the file, but now they can:

```js
/** @template T */
function f() { }
```

opposed to TS, which has no way to have a type parameter list that starts at position 0.

```ts
function f<T>() { }
```

Reply:
This might indicate that the synthetic type parameter list Pos should not be 0, though.

Reply:
Yep, all the Loc code in the reparser is barely past placeholder code. 
I need to put more thought into it, and this change might not be needed anymore.